mlflow/store/artifact/optimized_s3_artifact_repo.py

```diff
@@ -76,7 +77,7 @@ def _get_region_name(self):
         from botocore.exceptions import ClientError
 
         temp_client = _get_s3_client(
-            addressing_style="path",
+            addressing_style=MLFLOW_BOTO_CLIENT_ADDRESSING_STYLE.get(),
```

Comment:
```suggestion
            addressing_style=self._addressing_style,
```

Seems odd to not honor the passed-in addressing style (or use the default if none specified)